### PR TITLE
Fix Anycubic Chiron SD navigation from TFT

### PR DESCRIPTION
**Pull request by SwiftNick**
When browsing SD card contents, panel would just lock up.
`println("/")`  behaviour has also changed.
now using
```
write('/')
println()
```

### CODE REVIEW

**Comment by rhapsodyv** [comment]
it seems we have a problem here. Why println(“/“) doesn’t work anymore?

@X-Ryl669

**Comment by rhapsodyv** [comment]
I added the "Don't Merge" label because seems this PR are showing us something broken on our serial code.

**Comment by X-Ryl669** [comment]
I've tested println and it's working for me. So this is something else going on here.

File: Marlin/src/lcd/extui/lib/anycubic_chiron/chiron_tft.cpp [hunk 1]
```diff
@@ -292,7 +292,7 @@ namespace Anycubic {
     #if ACDEBUG(AC_SOME)
       serialprintPGM(str);
     #endif
-    while (const char c = pgm_read_byte(str++)) TFTSer.print(c);
+    while (const char c = pgm_read_byte(str++)) TFTSer.write(c);
```
Comment (X-Ryl669) [inline Marlin/src/lcd/extui/lib/anycubic_chiron/chiron_tft.cpp:295]: I'm almost sure that's the only part that was really required in this patch. `print(char)` prints the value as a number while write output it. `println(const char*)` and `print(const char*)` have not changed behavior

File: Marlin/src/sd/usb_flashdrive/lib-uhs3/UHS_host/UHS_host_INLINE.h [hunk 2]
```diff
@@ -984,7 +984,7 @@ uint8_t UHS_USB_HOST_BASE::ctrlReq(uint8_t addr, uint64_t Request, uint16_t nbyt
         //bool direction = bmReqType & 0x80; //request direction, IN or OUT
         uint8_t rcode = 0;
 
-        //        Serial.println("");
+        //Serial.println();
```
Comment (X-Ryl669) [inline Marlin/src/sd/usb_flashdrive/lib-uhs3/UHS_host/UHS_host_INLINE.h:987]: Ok

File: Marlin/src/lcd/extui/lib/anycubic_chiron/chiron_tft.cpp [hunk 3]
```diff
@@ -840,7 +840,7 @@ namespace Anycubic {
         // Ignore request if printing
         //if (isPrinting()) break;
         //injectCommands_P(PSTR("M500\nM420 S1\nG1 Z10 F240\nG1 X0 Y0 F6000"));
-        //TFTSer.println("");
+        //TFTSer.println();
```
Comment (X-Ryl669) [inline Marlin/src/lcd/extui/lib/anycubic_chiron/chiron_tft.cpp:843]: Ok

File: Marlin/src/lcd/extui/lib/anycubic_chiron/chiron_tft.cpp [hunk 4]
```diff
@@ -578,8 +578,8 @@ namespace Anycubic {
       case 15:   // A15 Resuming from outage
         if (printer_state == AC_printer_resuming_from_power_outage) {
           // Need to home here to restore the Z position
-          injectCommands(AC_cmnd_power_loss_recovery);
-          injectCommands("M1000");  // home and start recovery
+          injectCommands_P(AC_cmnd_power_loss_recovery);
```
Comment (X-Ryl669) [inline Marlin/src/lcd/extui/lib/anycubic_chiron/chiron_tft.cpp:581]: Ok

File: Marlin/src/lcd/extui/lib/anycubic_chiron/chiron_tft.cpp [hunk 5]
```diff
@@ -305,7 +305,7 @@ namespace Anycubic {
         SERIAL_EOL();
       #endif
     }
-    TFTSer.println("");
+    TFTSer.println();
```
Comment (X-Ryl669) [inline Marlin/src/lcd/extui/lib/anycubic_chiron/chiron_tft.cpp:308]: Ok, for optimization.

File: Marlin/src/lcd/extui/lib/anycubic_chiron/FileNavigator.cpp [hunk 6]
```diff
@@ -110,11 +110,12 @@ namespace Anycubic {
...
       TFTSer.println(filelist.shortFilename());
       TFTSer.print(filelist.shortFilename());
-      TFTSer.println("/");
+      TFTSer.write('/');
+      TFTSer.println();
     }
     else {
       // Logical Name
-      TFTSer.print("/");
+      TFTSer.write('/');
```
Comment (X-Ryl669) [inline Marlin/src/lcd/extui/lib/anycubic_chiron/FileNavigator.cpp:117]: Ok for optimization. Should not break if left as-is

**Comment by X-Ryl669** [comment]
I think the OP was a bit overzealous replacing all print with write. There only one case where it was required (see review). I don't think anything is broken here.

File: Marlin/src/lcd/extui/lib/anycubic_chiron/FileNavigator.cpp [hunk 7]
```diff
@@ -110,11 +110,12 @@ namespace Anycubic {
       //TFTSer.print(currentfoldername);
       TFTSer.println(filelist.shortFilename());
       TFTSer.print(filelist.shortFilename());
-      TFTSer.println("/");
+      TFTSer.write('/');
```
Comment (X-Ryl669) [inline Marlin/src/lcd/extui/lib/anycubic_chiron/FileNavigator.cpp]: Can you try without this change ? I'm almost sure it's working fine (tried here on my board).

**Comment by thinkyhead** [comment]
Since it's late in the day I will merge the uncontroversial parts of this PR now.

**Comment by rhapsodyv** [comment]
> I think the OP was a bit overzealous replacing all print with write. There only one case where it was required (see review). I don't think anything is broken here. 

Thanks for your review!

**Comment by SwiftNick** [comment]
@X-Ryl669 
After PR21152 was merged, another Chiron user advised SD folder navigation still wasn't working.
I stuck a serial logger on the panel, and there was a bad character at the point of the `println("/")`.
Changing to a `write('/') println()` fixed the issue, for both mine and the other Chiron.

Following your comments I have just tried reverting to the print("/") and it works, but it didn't before?!?
Odd...

**Comment by X-Ryl669** [comment]
I wonder if the bad character was left from the print(char as number) instead of the write(char) call. Also, I've made some changes to the serial redirection code that was merged meanwhile, so if the other user was in a multiserial setup, it's possible the char was coming/leaking from the other serial port, but it should be fixed by now.

**Comment by SwiftNick** [comment]
Possibly,
He was using the same configs and extui interface that I wrote for the Chiron so the setup would have been identical.
It's all working ok now!

Now working on the FileNavigator to support the new TFT panel Anycubic are shipping. 
It only shows strings that contain '.GCO' or '.gcode', so the PC like folder browsing for SD I spent ages on,
wont work that way on the new panels. :-(

**Pull request merged**